### PR TITLE
updates for api from staging maven repository and corresponding signature files.

### PR DESCRIPTION
**Pull request by gurunrao**
updates for latest api from staging maven repository and corresponding signature files.
    jakarta.jms-api - https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/jms/jakarta.jms-api
    jakarta.xml.bind-api - https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/xml/bind/jakarta.xml.bind-api/
    jakarta.faces-api - https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/faces/jakarta.faces-api/
    jakarta.persistence-api - https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/persistence/jakarta.persistence-api/
    jakarta.transaction-api - https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/transaction/jakarta.transaction-api/
    jakarta.resource-api - https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/resource/jakarta.resource-api/
    jakarta.json.bind-api - https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/json/bind/jakarta.json.bind-api/
    jakarta.servlet.jsp.jstl-api - https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/servlet/jsp/jstl/jakarta.servlet.jsp.jstl-api/
    webservices-api-osgi - https://jakarta.oss.sonatype.org/content/repositories/staging/org/glassfish/metro/webservices-api-osgi/
    webservices-api - https://jakarta.oss.sonatype.org/content/repositories/staging/org/glassfish/metro/webservices-api/
    webservices-tools - https://jakarta.oss.sonatype.org/content/repositories/staging/org/glassfish/metro/webservices-tools/

### CODE REVIEW

File: docker/pom.xml [hunk 1]
```diff
@@ -47,22 +47,22 @@
...
         <jsonp-api.version>2.1.0</jsonp-api.version>
-        <json.bind-api.version>3.0.0-RC1</json.bind-api.version>
+        <json.bind-api.version>3.0.0</json.bind-api.version>
         <!-- Jakarta Server Pages -->
         <jsp-api.version>3.1.0</jsp-api.version>
         <!-- Jakarta Standard Tag Library -->
-        <jstl-api.version>2.0.0</jstl-api.version>        
+        <jstl-api.version>3.0.0-RC1</jstl-api.version>        
         <!-- CDI -->
         <cdi-api.version>3.0.0</cdi-api.version>
```
Comment (scottmarlow) [inline docker/pom.xml:67]: looks like we need to update cdi-api to [4.0.0-RC5](https://search.maven.org/artifact/jakarta.enterprise/jakarta.enterprise.cdi-api/4.0.0-RC5/jar) but there are additional changes needed for CDI, as:
```
bash docker/build_signatures.sh
...
Warning: The return type jakarta.enterprise.lang.model.declarations.DeclarationInfo can't be resolved 
Warning: The return type jakarta.enterprise.lang.model.declarations.DeclarationInfo can't be resolved 
Warning: The return type jakarta.enterprise.lang.model.declarations.DeclarationInfo can't be resolved 
Warning: The return type jakarta.enterprise.lang.model.declarations.DeclarationInfo can't be resolved 
Warning: class jakarta.enterprise.lang.model.types.WildcardType not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.types.ParameterizedType not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.types.ArrayType not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.types.PrimitiveType not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.AnnotationInfo not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.types.Type not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.types.VoidType not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.declarations.DeclarationInfo not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.declarations.MethodInfo not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.AnnotationTarget not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.declarations.ClassInfo not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.types.ClassType not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.declarations.FieldInfo not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.types.PrimitiveType$PrimitiveKind not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.AnnotationMember not found. Please, add required jar or directory to the classpath.
Warning: class jakarta.enterprise.lang.model.declarations.ParameterInfo not found. Please, add required jar or directory to the classpath.
Selected by -Package: 211 classes
Filtered out by -Exclude: 30 classes
Written to sigfile: 177 classes(and 20 inner classes)
STATUS:Failed.16 errors
...
```

Comment (gurunrao) [reply docker/pom.xml:67]: latest CDI API jars are not present at staging - https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/enterprise/cdi-api/

Comment (scottmarlow) [reply docker/pom.xml:67]: > latest CDI API jars are not present at staging - https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/enterprise/cdi-api/

I think that `4.0.0-RC5` should be available.

Comment (scottmarlow) [reply docker/pom.xml:67]: > Also not able to upgrade to latest JMS API (https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/jms/jakarta.jms-api/) due to following error: `[ERROR] Failed to execute goal on project dependency_copy: Could not resolve dependencies for project jakarta:dependency_copy:jar:1: Failed to collect dependencies at jakarta.jms:jakarta.jms-api:jar:3.1.0: Failed to read artifact descriptor for jakarta.jms:jakarta.jms-api:jar:3.1.0: org.eclipse.ee4j.jms:jms-parent:pom:3.1.0 was not found in https://jakarta.oss.sonatype.org/content/repositories/staging/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of jakarta-snapshots has elapsed or updates are forced -> [Help 1] [ERROR] `

The same reproduces locally for me:
```
Downloading from central: https://repo.maven.apache.org/maven2/org/eclipse/ee4j/jms/jms-parent/3.1.0/jms-parent-3.1.0.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.727 s
[INFO] Finished at: 2022-02-28T11:45:30-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project dependency_copy: Could not resolve dependencies for project jakarta:dependency_copy:jar:1: Failed to collect dependencies at jakarta.jms:jakarta.jms-api:jar:3.1.0: Failed to read artifact descriptor for jakarta.jms:jakarta.jms-api:jar:3.1.0: Could not find artifact org.eclipse.ee4j.jms:jms-parent:pom:3.1.0 in jakarta-staging (https://jakarta.oss.sonatype.org/content/repositories/staging/) -> [Help 1]
```

Comment (scottmarlow) [reply docker/pom.xml]: Email sent to jms-dev ml for ^

Comment (gurunrao) [reply docker/pom.xml]: moved PR to DRAFT and will wait for jms api resolution.

Comment (OndroMih) [reply docker/pom.xml]: Thanks for reporting this. This is fixed in the source code, I'm working on staging a new version of the API module that doesn't depend on the messaging parent module.

Comment (scottmarlow) [reply docker/pom.xml]: I had to update my local maven repo to remove `jakarta.jms:jakarta.jms-api:jar:3.1.0` artifacts and then ran `bash docker/build_signatures.sh` again which looked good with the JMS change.  I pushed https://github.com/scottmarlow/jakartaee-tck/tree/api-update-2 to update the JMS + CDI SPEC API version.

**Review by scottmarlow — APPROVED** [review]

**Pull request merged**